common/array.h

```diff
@@ -107,7 +110,7 @@ class Array {
 	}
 
 	/**
-	 * Constructs an array using list initialization.
+	 * Construct an array using list initialization.
```

Comment:
Maybe we should add an example to clarify what list initialisation is?
For example
`Common::Array<int> myArray = {1, 7, 42};`
constructs an array with 3 elements whose values are 1, 7, and 42 respectively.

Also since currently C++11 support can be disabled when running configure, we want to indicate that this constructor is only available when C++11 support is enabled.